src/Moq/ActionObserver.cs

```diff
@@ -165,12 +158,23 @@ bool CanDistribute(int msi, int asi)
 					}
 				}
 
-				// Finally box all values that need to be boxed:
+				// Finally, add explicit type casts (aka `Convert` nodes) where necessary:
```

Comment:
As above, it might be good to document the exact rules of how the C# compiler handles this as tests.